tests/server.test.ts

```diff
@@ -0,0 +1,64 @@
+import {assert} from "chai";
+import net from "net";
+import HobbitsP2PNetwork, {HobbitsOpts} from "../src";
+
+describe("Server", () => {
+  const ctx: HobbitsOpts = {
+    networkId: 1,
+    chainId: 1,
+    state: {
+      latestFinalizedRoot: Buffer.alloc(0),
+      latestFinalizedEpoch: 1,
+      latestFinalizedSlot: 1
+    }
+  };
+  // const server2 = new HobbitsP2PNetwork({...ctx, port: 9001, bootnodes: ["127.0.0.1:9000"]});
+
+  describe("Startup/Shutdown", async () => {
+    const server = new HobbitsP2PNetwork(ctx);
+
+    it("The server should not be running", async () => {
+      const status: boolean = server.running;
+      assert.isFalse(status, `Server should have returned false, but it returned with ${status}`);
+    });
+
+    it("The server should be running", async () => {
+      await server.start();
+      const status: boolean = server.running;
+      assert.isTrue(status, `Server should have returned true, but it returned with ${status}`)
+    });
+
+    it("The server should shutdown", async () => {
+      await server.stop();
+      const status: boolean = server.running;
+      assert.isFalse(status, `Server should have returned false, but it returned with ${status}`);
+    })
+  });
+
+  describe("Connections", () => {
+    const server = new HobbitsP2PNetwork(ctx);
+
+    before(async () => {
+      await server.start();
+    });
+
+    it("Server should have 0 connections", async () => {
+      // Check the total amount before connecting
+      const peersBefore: number = server.getTotalPeers();
+      assert.strictEqual(peersBefore, 0, `Expected total count to be 0 but got ${peersBefore}`);
+
+      describe("Connect to server", () => {
+        before(async () => {
+          await net.createConnection({port: 9000});
+        });
+        
+        // Check total connections after the before script runs
+        it("It should have 1 connection", () => {
```

Comment:
@wemeetagain @mpetrunic Can one of you have a look at this, im running into an async issue where the server doesn't have time to update itself. The `before` doesn't seem to be working

Reply:
I think mocha executes all befores (even nested ones) and then stars nested ones. Not sure if that helps.
https://gist.github.com/harto/c97d2fc9d0bfaf20706eb2acbf48c908

Reply:
@GregTheGreek I suspect this problem could be a result of putting a `before` *inside* of an `it`, but I can try and poke around with that if needed.

Reply:
Well the problem im trying to solve is that `createConnection` has asyc issues, where its finished executing but doesn't actually finish connecting so the next test fails because the connection hasn't finished on the server

Reply:
`net.createConnection` doesn't return promise, that's why await doesn't work. Use this

Use something like this:
```javascript
await new Promise((resolve) => {
   const client = net.createConnection({port: 9000});
   client.on('connect', resolve);
})
```

Make sure you close connection after tests otherwiseyour mocha will never finish